### DISCUSSION TITLE
Although multiple documents are processed only two of them are analyzed/used by the model

### DISCUSSION THREAD

**Discussion by ArjenR** [SecureAI-Tools/SecureAI-Tools#65] [Q&A]
If you upload more than two PDF I noticed only the same two of them are used by the model for replying to questions. When asked it also will say it only knows of 2 documents and when asked questions specific to the other documents it will reply to have no knowledge thereof.

**Comment by JayNakrani** [comment]
When answering, it finds the **top K document-chunks out of all documents** in the collection and then constructs a response based on those two chunks. Please change the value of `DOCS_RETRIEVAL_K` env variable if you want it to take more document-chunks into account

**Reply by GitHub-Noscapine** [reply]
I uploaded only one PDF and chatted with that PDF. Results were not bad. I then uploaded 400+ PDFs and asked the same questions as when I uploaded only one PDF. The results were very poor. It could not refer to the single PDF I had tested when being part of a collection of 400+ PDFs, even if the title of that single PDF was in the question I asked. I tested the effects of changing the DOCS_RETRIEVAL_K variable (values between 2 and 100). Not much better. What else can be adjusted in the env variable to get better results ?
